rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -94,12 +94,12 @@ unsafe fn mark_compact<M: Memory, SetHp: Fn(u32)>(
 
     mark_static_roots(mem, static_roots, heap_base);
 
-    if (*closure_table_ptr_loc).unskew() >= heap_base as usize {
-        // TODO: No need to check if closure table is already marked
-        mark_object(mem, *closure_table_ptr_loc, heap_base);
-        // Similar to `mark_root_mutbox_fields`, `closure_table_ptr_loc` is in static heap so it
-        // will be readable when we unthread closure table
-        thread(closure_table_ptr_loc);
+    if (*continuation_table_ptr_loc).unskew() >= heap_base as usize {
```

Comment:
@osa1 Just to double check, your gc of the closure/continuation table doesn't assume anything about the Motoko object stored there right? It's just that they used to be scalars, closures or 2-element tuples and now may be scalars or 2 or 3 element tuples after @ggreif's recent change.

Reply:
@crusso From gc point of view closure table is just a heap object that is also a root. We don't make any assumptions on the type of it, or how many elements it has, or whether elements are scalar or pointers.